eventsStats.js

```diff
@@ -1,108 +1,142 @@
+          homeDeposit.returnValues.recipient === foreignDeposit.returnValues.from &&
+          homeDeposit.returnValues.value === foreignDeposit.returnValues.value
+        )
+      }).length === 0
+    )
   }
 }
-function compareTransferForeign(home){
-  return function(foreignDeposit){
-    return home.filter(function(homeDeposit){
-      return foreignDeposit.transactionHash === homeDeposit.returnValues.transactionHash
-        && foreignDeposit.returnValues.from === homeDeposit.returnValues.recipient &&
-        foreignDeposit.returnValues.value === homeDeposit.returnValues.value
-    }).length === 0;
+function compareTransferForeign(home) {
+  return foreignDeposit => {
+    return (
+      home.filter(homeDeposit => {
+        return (
+          foreignDeposit.transactionHash === homeDeposit.returnValues.transactionHash &&
+          foreignDeposit.returnValues.from === homeDeposit.returnValues.recipient &&
+          foreignDeposit.returnValues.value === homeDeposit.returnValues.value
+        )
+      }).length === 0
+    )
   }
 }
 
-async function main(){
+async function main() {
   try {
-    const isErcToErcMode = await existsErc677Token(web3Home, HOME_ERC_ABI, HOME_BRIDGE_ADDRESS)
-    logger.debug("isErcToErcMode", isErcToErcMode)
-    const HOME_ABI = isErcToErcMode ? HOME_ERC_ABI : HOME_NATIVE_ABI
-    const FOREIGN_ABI = isErcToErcMode ? FOREIGN_ERC_ABI : FOREIGN_NATIVE_ABI
-    const homeBridge = new web3Home.eth.Contract(HOME_ABI, HOME_BRIDGE_ADDRESS);
-    const foreignBridge = new web3Foreign.eth.Contract(FOREIGN_ABI, FOREIGN_BRIDGE_ADDRESS);
-    const erc20Contract = new web3Foreign.eth.Contract(ERC20_ABI, POA20_ADDRESS)
-    logger.debug("calling homeBridge.getPastEvents('UserRequestForSignature')");
-    let homeDeposits = await homeBridge.getPastEvents('UserRequestForSignature', {fromBlock: HOME_DEPLOYMENT_BLOCK});
-    logger.debug("calling foreignBridge.getPastEvents('RelayedMessage')");
-    let foreignDeposits = await foreignBridge.getPastEvents('RelayedMessage', {fromBlock: FOREIGN_DEPLOYMENT_BLOCK});
-    logger.debug("calling homeBridge.getPastEvents('AffirmationCompleted')");
-    let homeWithdrawals = await homeBridge.getPastEvents('AffirmationCompleted', {fromBlock: HOME_DEPLOYMENT_BLOCK});
-    logger.debug("calling foreignBridge.getPastEvents('UserRequestForAffirmation')");
-    let foreignWithdrawals = isErcToErcMode
-      ? await erc20Contract.getPastEvents('Transfer', {fromBlock: FOREIGN_DEPLOYMENT_BLOCK, filter: { to: FOREIGN_BRIDGE_ADDRESS }})
-      : await foreignBridge.getPastEvents('UserRequestForAffirmation', {fromBlock: FOREIGN_DEPLOYMENT_BLOCK})
+    const homeErcBridge = new web3Home.eth.Contract(HOME_ERC_TO_ERC_ABI, HOME_BRIDGE_ADDRESS)
+    const bridgeModeHash = await homeErcBridge.methods.getBridgeMode().call()
+    const bridgeMode = decodeBridgeMode(bridgeModeHash)
+    const isErcToErcMode = bridgeMode === BRIDGE_MODES.ERC_TO_ERC_MODE
```

Comment:
In order to proper handle ERC20-To-Native events this line should be:
 `const isErcToErcMode = bridgeMode === BRIDGE_MODES.ERC_TO_ERC_MODE || bridgeMode === BRIDGE_MODES.ERC_TO_NATIVE`.

Reply:
Done.